cms/static/js/views/modals/course_outline_modals.js

```diff
@@ -822,6 +822,74 @@ define(['jquery', 'backbone', 'underscore', 'gettext', 'js/views/baseview',
         }
     });
 
+    DiscussionEditor = AbstractEditor.extend({
+        templateName: 'discussion-editor',
+        className: 'edit-discussion',
+        afterRender: function() {
+            AbstractEditor.prototype.afterRender.call(this);
+            this.setStatus(this.currentValue());
+        },
+
+        currentValue: function() {
+            var discussion_enabled = this.model.get('discussion_enabled');
+            if (discussion_enabled === 'partially_enabled') {
+                return discussion_enabled
+            }
+            return discussion_enabled === true || discussion_enabled === 'enabled';
+        },
+
+        hasChildren: function() {
+            var child_info = this.model.get('child_info');
+            if (child_info && child_info.children) {
+                return Object.keys(child_info.children).length > 0;
+            }
+            return false;
+        },
+
+        setStatus: function(value) {
+            if (value === 'partially_enabled') {
+                this.$('#discussion_enabled').prop('indeterminate', true);
+            } else {
+                this.$('#discussion_enabled').prop('checked', value);
+            }
+        },
+
+        isIndeterminate: function() {
```

Comment:
I suspect this (and other changes) are related to and/or may change following the discussion of simplifying the UI indicators, so unless you tell me otherwise, I'm going to punt on further reviewing this file for now.